[Merged by Bors] - refactor(linear_algebra/bilinear_form): Change namespace of is_refl, is_symm, and is_alt

mcdoll opened the pull request:
The propositions `is_refl`, `is_symm`, and `is_alt` are now in the
namespace `bilin_form`. Moreover, `is_sym` is now renamed to `is_symm`.



---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

mcdoll committed:
refactor(linear_algebra/bilinear_form): Change namespace of is_refl, is_symm, and is_alt

The propositions `is_refl`, `is_symm`, and `is_alt` are now in the
namespace bilin_form. Moreover, `is_sym` is now renamed to `is_symm`.

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,14 +948,18 @@ end to_matrix
-variable (H : is_refl B)
+end bilin_form
+
+namespace refl_bilin_form
```
I think this namespace should change to `bilin_form.is_refl`, and same for the alt / symm namespaces

mcdoll replied:
there are several instances of the `is_refl` Lemma, what should we do about these? Rename to `symm_is_refl` and `alt_is_refl`?

eric-wieser replied:
They'll be in different namespaces still; the full names should be `bilin_form.is_symm.is_refl` and `bilin_form.is_alt.is_refl`

mcdoll committed:
refactor(linear_algebra/bilinear_form): Rename namespaces

The namespaces refl_bilin_form is changed to bilin_form.is_refl and same
for alt_bilin_form and sym_bilin_form

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,47 @@ end to_matrix
+
+namespace bilin_form.is_refl
+
+open bilin_form.is_refl bilin_form
```
```suggestion
namespace is_refl
```

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,47 @@ end to_matrix
-end refl_bilin_form
-
-namespace sym_bilin_form
+end bilin_form.is_refl
```
```suggestion
end is_refl
```

mcdoll replied:
Ah thanks, I did not know that. I should really read Theorem Proving in Lean..

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,47 @@ end to_matrix
+end bilin_form.is_refl
 
-open sym_bilin_form bilin_form
+namespace bilin_form
```
```suggestion
```

eric-wieser reviewed (commented):
Instead of writing:
```lean
namespace foo

-- foo.stuff

end foo

namespace foo.bar
open foo

-- foo.bar.stuff

end foo.bar
```
you can write the shorter:
```lean
namespace foo

-- foo.stuff

namespace bar

-- foo.bar.stuff

end bar
end foo
```

See my suggestions below for a partial example, hopefully enough to get the idea.

mcdoll committed:
simplified namespaces

eric-wieser commented on src/linear_algebra/quadratic_form/basic.lean:
```diff
@@ -462,7 +462,7 @@ end
 
 namespace quadratic_form
-open bilin_form sym_bilin_form
+open bilin_form bilin_form.is_symm
```
I would guess you don't need this open:
```suggestion
open bilin_form
```

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,37 @@ end to_matrix
+
+variable (H : B.is_symm)
 
 lemma sym (x y : M) : B x y = B y x := H x y
```
```suggestion
lemma symm (x y : M) : B x y = B y x := H x y
```
or perhaps even
```suggestion
protected lemma eq (x y : M) : B x y = B y x := H x y
```
to match `commute.eq`

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,37 @@ end to_matrix
+  is_ortho B x y ↔ is_ortho B y x := H.is_refl.ortho_sym
 
-lemma is_sym_iff_flip' [algebra R₂ R] : is_sym B ↔ flip_hom R₂ B = B :=
+lemma is_sym_iff_flip' [algebra R₂ R] : B.is_symm ↔ flip_hom R₂ B = B :=
```
```suggestion
lemma is_symm_iff_flip' [algebra R₂ R] : B.is_symm ↔ flip_hom R₂ B = B :=
```

eric-wieser replied:
ACtually, this should probably be in the `bilin_form` namespace, so that it's `bilin_form.is_symm_iff_flip'` not `bilin_form.is_symm.is_symm_iff_flip'.

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,37 @@ end to_matrix
-lemma is_refl : refl_bilin_form.is_refl B := λ x y H1, H x y ▸ H1
+lemma is_refl : B.is_refl := λ x y H1, H x y ▸ H1
 
 lemma ortho_sym {x y : M} :
```
```suggestion
lemma is_ortho_comm {x y : M} :
```
matches the name `eq_comm`

eric-wieser reviewed (commented):
This looks great, thanks! Just a few more renames to get rid of `sym`s

mcdoll committed:
consistency

mcdoll committed:
make the linter happy

eric-wieser commented on src/ring_theory/trace.lean:
```diff
@@ -152,7 +152,7 @@ variables {S}
 @[simp] lemma trace_form_apply (x y : S) : trace_form R S x y = trace R S (x * y) := rfl
 
-lemma trace_form_is_sym : sym_bilin_form.is_sym (trace_form R S) :=
+lemma trace_form_is_sym : (trace_form R S).is_symm :=
```
```suggestion
lemma trace_form_is_symm : (trace_form R S).is_symm :=
```

mcdoll committed:
Update src/ring_theory/trace.lean

Co-authored-by: Eric Wieser <wieser.eric@gmail.com>

eric-wieser commented on src/linear_algebra/quadratic_form/basic.lean:
```diff
@@ -498,7 +498,7 @@ variables (Q : quadratic_form R M) (S)
   associated_hom S Q x y = ⅟2 * (Q (x + y) - Q x - Q y) := rfl
 
-lemma associated_is_sym : is_sym (associated_hom S Q) :=
+lemma associated_is_sym : (associated_hom S Q).is_symm :=
```
```suggestion
lemma associated_is_symm : (associated_hom S Q).is_symm :=
```

eric-wieser commented on src/linear_algebra/quadratic_form/basic.lean:
```diff
@@ -801,7 +802,7 @@ begin
       finrank_span_singleton (ne_zero_of_not_is_ortho_self x hx)] at hd,
   let B' := B.restrict (B.orthogonal $ K ∙ x),
-  obtain ⟨v', hv₁⟩ := ih (B.restrict_sym hB₂ _ : sym_bilin_form.is_sym B') (nat.succ.inj hd),
+  obtain ⟨v', hv₁⟩ := ih (B.restrict_sym hB₂ _ : B'.is_symm) (nat.succ.inj hd),
```
We should rename `restrict_sym` to `restrict_symm` too

mcdoll committed:
more sym -> symm

mcdoll committed:
Merge branch 'bilinear' of github.com:leanprover-community/mathlib into bilinear

mcdoll committed:
Update src/linear_algebra/quadratic_form/basic.lean

Co-authored-by: Eric Wieser <wieser.eric@gmail.com>

mcdoll committed:
more sym -> symm

mcdoll committed:
Merge branch 'bilinear' of github.com:leanprover-community/mathlib into bilinear

eric-wieser commented on src/linear_algebra/bilinear_form.lean:
```diff
@@ -948,39 +948,39 @@ end to_matrix
-open refl_bilin_form bilin_form
+namespace bilin_form
 
 /-- The proposition that a bilinear form is reflexive -/
 def is_refl (B : bilin_form R M) : Prop := ∀ (x y : M), B x y = 0 → B y x = 0
```
Out of scope for this PR, but it would be good to mention these definitions in the module docstring at the top of the file.
